### PR TITLE
[wip] new structure

### PR DESCRIPTION
**Pull request by pi0**
This PR changes lib directory structure in a more modular one. One step towards modular packages and better ESM support.

Status:
- [x] Marge tests/lib/docs into src
- [x] Update builds
- [x] Update docs
- [x] Get search working
- [x] Update tests (optional)
- [ ] Tree shakeable esm builds (optional)

I think optional labeled parts can be non-blocking for 1.0.0 release (and merging this branch twice for faster release)

The new structure can be reviewed from [branch](https://github.com/bootstrap-vue/bootstrap-vue/tree/feat/new-structure/src). Much better than github diff :))

### CODE REVIEW

**Comment by mosinve** [comment]
Hm...tests wont fix... 
```
({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import { loadFixture, testVM } from "../../utils/helpers";
                                                                                             ^^^^^^
    SyntaxError: Unexpected token import
```
Will try to deal with'em a bit later

**Comment by pi0** [comment]
@mosinve I think this is because the lack of `.babelrc` and jest won't understand that :)) Thnks for your help, right now i'm focused on fixing docs. Will come back to help fixing tests too.

File: docs/utils/index.js [hunk 1]
```diff
@@ -90,3 +69,40 @@ export function importAll(r) {
...
+
+// Return an element's offset wrt document element
+// https://j11y.io/jquery/#v=git&fn=jQuery.fn.offset
+export function offsetTop(el) {
+    if (!el.getClientRects().length) {
+        return 0;
+    }
+    const bcr = el.getBoundingClientRect();
+    const win = el.ownerDocument.defaultView;
+    return bcr.top + win.pageYOffset;
```
Comment (tmorehouse) [inline docs/utils/index.js]: Where does `win` get defined?

Comment (pi0) [reply docs/utils/index.js]: Don't know just refactored old docs :))) What should it be?

Comment (tmorehouse) [reply docs/utils/index.js]: window :-)

Must have been in there somewhere originally (or missed) when I refactored it.  Although it appears to be working...

**Comment by tmorehouse** [comment]
Should `container` be moved into `layout` along side `col` and `row`?

**Comment by pi0** [comment]
It is also a dep of the carousel. But seems a good idea.

**Comment by tmorehouse** [comment]
Hmmm I didn't think carousel used b-container...

File: package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "url": "git+https://github.com/bootstrap-vue/bootstrap-vue.git"
   },
   "files": [
-    "lib",
+    "src",
     "dist",
     "nuxt"
```
Comment (tmorehouse) [inline package.json]: I think `nuxt` has to change to `docs`

Comment (pi0) [reply package.json]: This nuxt refers to nuxt module not docs. I will deprecate `@nuxtjs/bootstrap-vue` in flavor of this internal so we can offer an always update integration module :)

**Comment by pi0** [comment]
Seems we are in a stable status now. Really thanks for your contribution @tmorhouse @mosinve ❤😌

File: package.json (hunk 2 above)
Comment (tmorehouse) [reply package.json]: Ah, ok...

**Pull request merged**